src/classes/RD_RecurringDonations_TEST2.cls

```diff
@@ -1026,94 +1026,140 @@ private with sharing class RD_RecurringDonations_TEST2 {
         }
     }
 
-    /* Commenting this out temporarily
-    static testMethod void testOpenEndedMonthlyEndDates(){
-        if (strTestOnly != '*' && strTestOnly != 'testOpenEndedMonthlyEndDates') return;
+    static testMethod void testOpenEndedMonthlyEndDates_28th(){
+        if (strTestOnly != '*' && strTestOnly != 'testOpenEndedMonthlyEndDates_28th') return;
 
         UTIL_CustomSettingsFacade.getRecurringDonationsSettingsForTest(
             new npe03__Recurring_Donations_Settings__c(
             npe03__Opportunity_Forecast_Months__c = 12,
-            npe03__Maximum_Donations__c = 50,
-            npe03__Open_Opportunity_Behavior__c = RD_RecurringDonations.RecurringDonationCloseOptions.Mark_Opportunities_Closed_Lost.name()        
-            ));             
-        
-        Account a = new Account();
-        a.Name = 'test Individual';
-        insert a;
-        
-        Contact c = new Contact();
-        c.FirstName = 'test';
-        c.LastName = 'contact';
-        c.AccountId = a.Id;
-        insert c;       
-        
-        Campaign camp = new Campaign(name = 'Campaign', isActive = true);
-        insert camp;
-        
-        list<npe03__Recurring_Donation__c> rdlist = new list<npe03__Recurring_Donation__c>();
+            npe03__Maximum_Donations__c = 50
+            ));
+
+        Contact c = new Contact(
+            FirstName = 'test',
+            LastName = 'contact'
+        );
+        insert c;
+        Account a = [SELECT Id FROM Account WHERE Id IN (SELECT AccountId FROM Contact WHERE Id = :c.Id) LIMIT 1];
+
+        // This should create Installments with a date of the 28th.
+        npe03__Recurring_Donation__c rd = new npe03__Recurring_Donation__c();
+        rd.Name = 'test28th';
+        rd.npe03__Contact__c = c.Id;
+        rd.npe03__Amount__c = 100;
+        rd.npe03__Installment_Period__c = system.label.npe03.RecurringDonationInstallmentPeriodMonthly;
+        rd.npe03__Date_Established__c = system.today().toStartOfMonth().addDays(27);
+        rd.npe03__Schedule_Type__c = system.label.npe03.RecurringDonationMultiplyValue;
+        rd.npe03__Open_Ended_Status__c = system.label.npe03.RecurringDonationOpenStatus;
+
```

Comment:
@force2b this looks great, I apologize for not mentioning this in the first review but I don't think we should rely on the default being false and should explicitly set - rd.Always_Use_Last_Day_Of_Month__c = false;... thoughts?

Reply:
We could, but then should we update all of the test methods in the class to do the same thing?

Reply:
Yea I thought about that too, do you know if customers can change the default on a managed field?

Reply:
Just checked. They can't change the default.

Reply:
Ok sounds good, I think it's ok, I know as a best practice they recommend not relying on default values but since it is in our control, should be fine.